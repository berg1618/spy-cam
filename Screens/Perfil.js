import { View, Text, TouchableOpacity, Image } from "react-native";
import FontAwesomeIcon from "react-native-vector-icons/FontAwesome";
import Header from "../Components/Header";
import CustomizedBar from "../Components/CustomizedBar";

const Perfil = () => {
  return (
    <View style={styles.container}>
      <CustomizedBar />
      <Header />
      <TouchableOpacity style={styles.circleButton}>
        <FontAwesomeIcon name='bell' size={40} color='#413C45' style={{ marginTop: 21, marginHorizontal: 24 }} />
      </TouchableOpacity>
      <Text style={styles.newTitle}>Perfil</Text>
      <View style={styles.line} />
      <View style={styles.contentContainer}>
        <View style={{ backgroundColor: '#FFFFFF', borderTopLeftRadius: 20, borderTopRightRadius: 20, borderBottomLeftRadius: 20, borderBottomRightRadius: 20, padding: 20, flex: 1 }}>
          <TouchableOpacity style={{ alignItems: 'center', justifyContent: 'center' }}>
            <Image source={require('../assets/iconrosto.png')} style={{ width: 80, height: 80 }} />
          </TouchableOpacity>
          <Text style={styles.anotherTitle}>Maria</Text>
          <View style={styles.anotherLine} />
        </View>
      </View>

    </View>

  );
};

const styles = {
  container: {
    flex: 1,
    backgroundColor: '#413C45',
  },
  newTitle: {
    color: '#fff',
    fontSize: 28,
    fontWeight: 'bold',
    textAlign: 'center',
    marginVertical: 20,
  },
  anotherTitle: {
    color: '#413C45',
    fontSize: 16,
    textAlign: 'center',
  },
  line: {
    borderBottomWidth: 2,
    borderBottomColor: '#fff',
    marginHorizontal: 100,
    marginVertical: -20,
  },
  anotherLine: {
    borderBottomWidth: 2,
    borderBottomColor: '#413C45',
    marginHorizontal: 115,
    marginVertical: 0,
  },
  contentContainer: {
    backgroundColor: '#fff',
    borderRadius: 10,
    margin: 40,
    marginVertical: 60,
    flex: 1,
    zIndex: -1,
  },
  circleButton: {
    position: 'absolute',
    bottom: 30,
    right: 20,
    width: 90,
    height: 90,
    borderRadius: 45,
    borderWidth: 1,
    borderColor: '#413C45',
    backgroundColor: '#FFFFFF',
  },

};

export default Perfil;
