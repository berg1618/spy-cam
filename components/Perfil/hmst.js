import React, { useState } from "react";
import { View, FlatList, Text, TouchableOpacity, Image, StyleSheet } from "react-native";
import { Ionicons } from "@expo/vector-icons";

const hmst = ({ navigation }) => {
  const [data, setData] = useState([{ id: "1", text: "Item 1" }, { id: "2", text: "Item 2" }]);

  const renderItem = ({ item }) => {
    const handleDeleteItem = (itemId) => {
      const updatedData = data.filter((item) => item.id !== itemId);
      setData(updatedData);
    };

    return (
      <View style={styles.itemContainer}>
        <TouchableOpacity style={styles.deleteButton} onPress={() => handleDeleteItem(item.id)}>
          <Ionicons name="close-circle" size={16} color="#413C45" />
        </TouchableOpacity>
        <TouchableOpacity style={styles.imageContainer}>
          <Image source={require('../../assets/iconrosto.png')} style={styles.image} />
        </TouchableOpacity>
        <Text style={styles.anotherTitle}>{item.text}</Text>
        <View style={styles.anotherLine} />
      </View>
    );
  };

  const handleAddItem = () => {
    const newItem = { id: Date.now().toString(), text: `Item ${data.length + 1}` };
    setData([...data, newItem]);
  };

  return (
    <View style={styles.container}>
      <FlatList
        data={data}
        renderItem={renderItem}
        keyExtractor={(item) => item.id}
        numColumns={2}
      />
      <TouchableOpacity onPress={handleAddItem} style={styles.addButton}>
        <Text style={styles.addButtonText}>+</Text>
      </TouchableOpacity>
      <Text
                onPress={() => {
                navigation.navigate("CadastrarRosto")
              }}
              style={styles.mensagem}>
                <Text>Go</Text>
              </Text>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
  },
  itemContainer: {
    position: "relative",
    flex: 1,
    justifyContent: "center",
    alignItems: "center",
    margin: 8,
    padding: 16,
    backgroundColor: "#f5f5f5",
    borderRadius: 20,
  },
  imageContainer: {
    alignItems: "center",
    justifyContent: "center",
  },
  image: {
    width: 80,
    height: 80,
  },
  anotherTitle: {
    color: "#413C45",
    fontSize: 16,
    textAlign: "center",
  },
  anotherLine: {
    borderBottomWidth: 2,
    borderBottomColor: "#413C45",
    marginHorizontal: 115,
    marginVertical: 8,
  },
  addButton: {
    position: "absolute",
    bottom: 16,
    right: 16,
    paddingHorizontal: 16,
    paddingVertical: 8,
    backgroundColor: "#007AFF",
    borderRadius: 8,
  },
  addButtonText: {
    color: "#FFFFFF",
    fontSize: 18,
  },
  deleteButton: {
    position: "absolute",
    top: 8,
    right: 8,
    width: 24,
    height: 24,
    borderRadius: 12,
    backgroundColor: "transparent",
    alignItems: "center",
    justifyContent: "center",
  },
});

export default hmst;
