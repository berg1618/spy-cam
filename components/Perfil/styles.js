import { StyleSheet } from 'react-native';

const styles = StyleSheet.create({
    container: {
        flex: 1,
        backgroundColor: '#413C45',
      },
      header: {
        flexDirection: 'row',
        alignItems: 'center',
        justifyContent: 'center',
        height: 150,
        backgroundColor: 'transparent',
        borderBottomWidth: 2,
        borderBottomColor: '#FFFFFF',
        position: 'relative',
      },
      title: {
        color: '#fff',
        fontWeight: 'bold',
        fontSize: 40,
        textAlign: 'center',
        position: 'absolute',
        bottom: 30,
        left: 0,
        right: 0,
      },
      menuIcon: {
        position: 'absolute',
        left: 140,
        fontSize: 40,
      },
      menuContainer: {
        position: 'absolute',
        top: 120,
        right: 0,
        backgroundColor: '#fff',
        padding: 20,
        borderRadius: 10,
      },
      menuItem: {
        fontSize: 18,
        color: '#000',
        marginBottom: 10,
      },
      newTitle: {
        color: '#fff',
        fontSize: 28,
        fontWeight: 'bold',
        textAlign: 'center',
        marginVertical: 20,
      },
      anotherTitle: {
        color: '#413C45',
        fontSize: 16,
        textAlign: 'center',
      },
      line: {
        borderBottomWidth: 2,
        borderBottomColor: '#fff',
        marginHorizontal: 100,
        marginVertical: -20,
      },
      anotherLine: {
        borderBottomWidth: 2,
        borderBottomColor: '#413C45',
        marginHorizontal: 115,
        marginVertical: 0,
      },
      contentContainer: {
        backgroundColor: '#fff',
        borderRadius: 10,
        margin: 40,
        marginVertical: 60,
        flex: 1,
        zIndex: -1,
        elevation: -1
      },
      circleButton: {
        position: 'absolute',
        bottom: 30,
        right: 20,
        width: 90,
        height: 90,
        borderRadius: 45,
        borderWidth: 1,
        borderColor: '#413C45',
        backgroundColor: '#FFFFFF',
      }
});

export default styles;