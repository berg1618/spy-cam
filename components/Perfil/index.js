import { View, Text, TouchableOpacity, Image } from 'react-native';
import FontAwesomeIcon from 'react-native-vector-icons/FontAwesome';
import styles from "./styles";
import Header from '../Header';
import CustomizedBar from '../CustomizedBar/CustomizedBar';
import PerfilFlatlist from '../PerfilFlatlist/PerfilFlatlist';

const Perfil = () => {
  return (
    <View style={styles.container}>
      <CustomizedBar />
      <Header />
      <TouchableOpacity style={styles.circleButton}>
        <FontAwesomeIcon name='bell' size={40} color='#413C45' style={{ marginTop: 21, marginHorizontal: 24 }} />
      </TouchableOpacity>
      <Text style={styles.newTitle}>Perfil</Text>
      <View style={styles.line} />
      <View style={styles.contentContainer}>
        <View style={{ backgroundColor: '#FFFFFF', borderTopLeftRadius: 20, borderTopRightRadius: 20, borderBottomLeftRadius: 20, borderBottomRightRadius: 20, padding: 20, flex: 1 }}>
          <TouchableOpacity style={{ alignItems: 'center', justifyContent: 'center' }}>
            <Image source={require('../../assets/iconrosto.png')} style={{ width: 80, height: 80 }} />
          </TouchableOpacity>
          <Text style={styles.anotherTitle}>VocÃª</Text>
          <View style={styles.anotherLine} />
          <PerfilFlatlist />
        </View>
      </View>

    </View>

  );
};

export default Perfil;