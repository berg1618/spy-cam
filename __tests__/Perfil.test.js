import React from 'react';
// import renderer from 'react-test-renderer';
import {render} from '@testing-library/react-native'
import Perfil from '../components/Perfil';

describe("teste do componente perfil", () => {
  test('Renderizando o componente', () => {
    render(<Perfil />)
  });
})
