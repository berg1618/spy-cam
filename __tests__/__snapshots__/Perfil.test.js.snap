// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testando o componente de perfil 1`] = `
<View
  style={
    {
      "backgroundColor": "#413C45",
      "flex": 1,
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "backgroundColor": "transparent",
        "borderBottomColor": "#FFFFFF",
        "borderBottomWidth": 2,
        "flexDirection": "row",
        "height": 150,
        "justifyContent": "center",
        "position": "relative",
      }
    }
  >
    <Text
      style={
        {
          "bottom": 30,
          "color": "#fff",
          "fontSize": 40,
          "fontWeight": "bold",
          "left": 0,
          "position": "absolute",
          "right": 0,
          "textAlign": "center",
        }
      }
    >
      Spy Cam
    </Text>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
    />
    <View
      collapsable={false}
      style={
        {
          "backgroundColor": "#fff",
          "borderRadius": 10,
          "padding": 20,
          "position": "absolute",
          "right": 0,
          "top": 120,
          "transform": [
            {
              "translateX": 300,
            },
          ],
        }
      }
    >
      <Text
        onPress={[Function]}
        style={
          {
            "color": "#000",
            "fontSize": 18,
            "marginBottom": 10,
          }
        }
      >
        Pagina inicial
      </Text>
      <Text
        onPress={[Function]}
        style={
          {
            "color": "#000",
            "fontSize": 18,
            "marginBottom": 10,
          }
        }
      >
        Notificacoes
      </Text>
      <Text
        style={
          {
            "color": "#000",
            "fontSize": 18,
            "marginBottom": 10,
          }
        }
      >
        Exemplo 3
      </Text>
      <Text
        style={
          {
            "color": "#000",
            "fontSize": 18,
            "marginBottom": 10,
          }
        }
      >
        Exemplo 4
      </Text>
    </View>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "backgroundColor": "#FFFFFF",
        "borderColor": "#413C45",
        "borderRadius": 45,
        "borderWidth": 1,
        "bottom": 30,
        "height": 90,
        "opacity": 1,
        "position": "absolute",
        "right": 20,
        "width": 90,
      }
    }
  />
  <Text
    style={
      {
        "color": "#fff",
        "fontSize": 28,
        "fontWeight": "bold",
        "marginVertical": 20,
        "textAlign": "center",
      }
    }
  >
    Perfil
  </Text>
  <View
    style={
      {
        "borderBottomColor": "#fff",
        "borderBottomWidth": 2,
        "marginHorizontal": 100,
        "marginVertical": -20,
      }
    }
  />
  <View
    style={
      {
        "backgroundColor": "#fff",
        "borderRadius": 10,
        "flex": 1,
        "margin": 40,
        "marginVertical": 60,
        "zIndex": -1,
      }
    }
  >
    <View
      style={
        {
          "backgroundColor": "#FFFFFF",
          "borderBottomLeftRadius": 20,
          "borderBottomRightRadius": 20,
          "borderTopLeftRadius": 20,
          "borderTopRightRadius": 20,
          "flex": 1,
          "padding": 20,
        }
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={false}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "justifyContent": "center",
            "opacity": 1,
          }
        }
      >
        <Image
          source={
            {
              "testUri": "../../../assets/iconrosto.png",
            }
          }
          style={
            {
              "height": 80,
              "width": 80,
            }
          }
        />
      </View>
      <Text
        style={
          {
            "color": "#413C45",
            "fontSize": 16,
            "textAlign": "center",
          }
        }
      >
        Maria
      </Text>
      <View
        style={
          {
            "borderBottomColor": "#413C45",
            "borderBottomWidth": 2,
            "marginHorizontal": 115,
            "marginVertical": 0,
          }
        }
      />
    </View>
  </View>
</View>
`;
